### PR TITLE
Use ExternalProject for googletest

### PR DESCRIPTION
**Pull request by timrae**
There were some TODO comments about using the official github directly from the cmake file, and due to some issues with cmake I can't use projects with submodules in my organization, so I went ahead and updated the cmake file.

### CODE REVIEW

**Comment by SRombauts** [comment]
Thanks a lot for provinding this! 
Please have a look at the build errors, something is not working as intended...

**Comment by timrae** [comment]
Yeah it's the yml files, I'll fix them now

**Comment by timrae** [comment]
Ugh, it looks like it's still failing on the Windows builds because I used `pthread`. I'll look into what the multi-platform lib should be

**Comment by timrae** [comment]
I can't see the error on travis, can you paste it here?

File: CMakeLists.txt [hunk 1]
```diff
@@ -249,22 +250,31 @@ endif (SQLITECPP_BUILD_EXAMPLES)
...
+        GIT_REPOSITORY https://github.com/google/googletest.git
+        INSTALL_COMMAND ""
+        CMAKE_ARGS -DCMAKE_C_COMPILER=$ENV{CC} -DCMAKE_CXX_COMPILER=$ENV{CXX}
+    )
 
     # add the unit test executable
     add_executable(SQLiteCpp_tests ${SQLITECPP_TESTS})
-    target_link_libraries(SQLiteCpp_tests gtest_main SQLiteCpp sqlite3)
+    add_dependencies(SQLiteCpp_tests googletest)
+    set(GTEST_MAIN_OBJ ${PREFIX_DIR}/src/googletest-build/googlemock/gtest/${LIB_PREFIX}gtest_main${LIB_SUFFIX})
```
Comment (timrae) [inline CMakeLists.txt]: @SRombauts 
Sorry I don't have access to a Windows C++ environment, so I don't know where these gtest binaries are being generated and what they're called there. On linux they're called `libgtest.a` and `libgtest_main.a`. Would you be able to check what's happening on Windows?

Comment (SRombauts) [inline CMakeLists.txt]: Sure, I'll try this at some point.

**Comment by SRombauts** [comment]
Closing, would need some serious rework

**Pull request closed without merging**